### ISSUE TITLE
Restore ability to use named volumes

### ISSUE THREAD

**Issue by JonathonReinhart** [JonathonReinhart/scuba#248]
Prior to #227, you could, as a not-officially-support feature map a named volume at a container path:
```yaml
volumes:
  /foo: foo-volume
```

```console
$ scuba --root --image debian:8.2 /bin/bash -c "echo yes > /foo/does-it-persist"
$ scuba --root --image debian:8.2 /bin/bash -c "cat /foo/does-it-persist"
yes
$ docker volume ls
DRIVER    VOLUME NAME
local     foo-volume
```

Since #227, this no longer works.
```
scuba: Config error: Relative path must start with ./ or ../: foo-volume
```

*Reported https://github.com/JonathonReinhart/scuba/pull/227#issuecomment-2005067223 by @haboustak.*

**Comment by JonathonReinhart** [comment]
From #227:

> To avoid ambiguity with the old behavior, relative paths must start with `./` or `../`.

And [the v2.12 docs](https://scuba.readthedocs.io/en/v2.12.0/configuration.html#conf-volumes)

> To avoid ambiguity, relative paths must start with `./` or `../`.

So I think we can easily restore the old behavior, using this required prefix to distinguish the two:
- If it starts with `/`, then it is a (host) absolute path.
- If it starts with `./` or `../`, then it is a (host) relative path.
- Otherwise:
  - If it contains a `/` it is an error.
  - Otherwise it is a volume name.

**NOTE:** This only applies to the simple form:
```yaml
volumes:
  /foo: foo-volume
```

In the complex form, the key is specifically called `hostpath`, so we shouldn't use that for a volume name. We should introduce another key -- mutually exclusive with `hostpath` -- perhaps just `name` since it is the name of a named volume:
```yaml
volumes:
  /foo:
    #hostpath: foo-volume    # Error
    name: foo-volume
```


See also:
- https://docs.docker.com/compose/compose-file/05-services/#volumes

**Comment by JonathonReinhart** [comment]
This is released in [v2.13.0](https://github.com/JonathonReinhart/scuba/releases/tag/v2.13.0).

Enjoy your Go module caches!
